### PR TITLE
4094 Enhance `ckpt_export` to save config files

### LINKED ISSUE #4094: Checkpoint Export Functionality Should Require Bundle Directory Path Only
**Is your feature request related to a problem? Please describe.**
Currently to export a bundle as a Torchscript file one has to specify the files to include individually, eg:

```bash
python -m monai.bundle ckpt_export network_def \
    --filepath foo.ts \
    --ckpt_file foo/models/model.pt \
    --meta_file foo/configs/metadata.json \
    --config_file foo/configs/inference.json
```

**Describe the solution you'd like**
I think the optimal behaviour is to specify export of a whole bundle directory which will include all the config files in the `configs` directory and use `models/model.pt` as the model weight file implicitly. The network definition `network_def` is in config files in the tutorials example  but perhaps it should be moved to the `metadata.json` file since there will really only be the one primary network definition per bundle. The actual command line to export would be :

```bash
python -m monai.bundle ckpt_export --bundle_path /path/to/bundle/dir --out_path foo.ts 
```

 (I would also suggest changing `filepath` to `out_path` to be clearer about what it is.)

**Describe alternatives you've considered**
The current command works just fine so this is a QoL improvement.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #4094 .

### Description
This PR enhanced the `ckpt_export` script to save config files separately.
And also added unit tests to load the TorchScript model with meta, configs.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/bundle/scripts.py [hunk 1]
```diff
@@ -415,12 +419,22 @@ def ckpt_export(
     # convert to TorchScript model and save with meta data, config content
     net = convert_to_torchscript(model=net)
 
+    extra_files: Dict = {}
+    for i in ensure_tuple(config_file_):
+        # split the filename and directory
+        _, filename = os.path.split(i)
```
Comment (ericspod) [inline monai/bundle/scripts.py:425]:
```suggestion
        filename = os.path.basename(i)
```

File: monai/bundle/scripts.py [hunk 2]
```diff
@@ -415,12 +419,22 @@ def ckpt_export(
...
+        # remove extension
+        filename, _ = os.path.splitext(filename)
```
Comment (ericspod) [inline monai/bundle/scripts.py:427]: Why not just remove this line? `filename` will contain the base name of the file which can be put directly into `extra_files`.

File: monai/bundle/scripts.py [hunk 3]
```diff
@@ -415,12 +419,22 @@ def ckpt_export(
...
+        if filename in extra_files:
+            raise ValueError(f"already have config in the extra files with same file name: {filename}.")
```
Comment (ericspod) [inline monai/bundle/scripts.py:429]:
```suggestion
            raise ValueError(f"Base filename '{filename}' is given multiple times in config file list")
```

The alternative is make sure the filenames are unique but that seems too complicated.

**Review by ericspod — CHANGES_REQUESTED** [review]
I think we can preserve the extensions of files we're saving with the minor change I'd suggest unless I'm missing something about why this was there originally.

File: monai/bundle/scripts.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/bundle/scripts.py:427]: Thanks for your suggestion, so you mean if the config is YAML file, we should `yaml.dumps()` to serialize it?
Now this PR always uses `json.dumps()` to serialize:
https://github.com/Project-MONAI/MONAI/pull/4159/files#diff-a715c9207f1e727d7a6ed7c6b25dfb57045d4429fc0b4eed3f5396e26ab34f80R430

Thanks.

File: monai/bundle/scripts.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/bundle/scripts.py]: Sure, updated in the latest commit.

Thanks.

File: monai/bundle/scripts.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/bundle/scripts.py]: Sure, updated in the latest commit.

Thanks.

File: monai/bundle/scripts.py (hunk 2 above)
Comment (Nic-Ma) [reply monai/bundle/scripts.py:427]: As discussed offline, let's save the configs without extension.

Thanks.

**Comment by Nic-Ma** [comment]
Hi @ericspod ,

Thanks for your review, I updated the PR according to your comments.
Could you please help review it again?

Thanks in advance.

**Review by ericspod — APPROVED** [review]

**Pull request merged**